### ISSUE TITLE
改了模型结构之后loss特别大，也降不下去

### ISSUE THREAD

**Issue by elory-y** [bubbliiiing/faster-rcnn-pytorch#85]
你好，我在backbone之后又增加了一个特征提取的模块，所以voc_weight_resnet的权重就用不了了，然后我使用了Resnet50的预训练权重，损失达到了10的33次方，调大学习率，调整学习率更新的方式也降不下去，请问是什么原因？有没有什么建议？谢谢

**Comment by bubbliiiing** [comment]
……我觉得你特征提取的模块可能有点问题啊兄弟

**Comment by dogdogpee** [comment]
我是只加载backbone在ImageNet上与训练的参数时也遇到了loss ~10^35的情况。把train.py里的
```
if pretrained:
        weights_init(model)
```
注释掉就好了.否则此段代码重置所有网络参数，加载的与训练参数就用不上了.

**Comment by bubbliiiing** [comment]
应该是if not pretrain的……我的问题

**Comment by dogdogpee** [comment]
改过来后虽然不爆炸，但在行人数据集上算的 train loss 只降了10%就开始波动了😭不知道咋回事 难道是因为我把resnet layer4也算到extractor里了，classifier只保留了avgpool？

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by bubbliiiing** [comment]
你没改过的代码损失大概多少，现在多少？

**Comment by dogdogpee** [comment]
没改时总的train loss是10^35，且不下降。改了以后第一个epoch的总train loss大约1.x，迭代三四十个epoch后在1.0震荡

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by bubbliiiing** [comment]
我说的是……如果直接下我这个代码，不做修改，用我提供的预训练

**Comment by dogdogpee** [comment]
是指load全模型预训练参数，然后freeze backbone训练吗？我还没试过这个诶 。你的意思是rpn参数初始化可能有问题？


Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by bubbliiiing** [comment]
是的。最好还是试一下，我觉得可能整个修改有点问题😂

**Comment by xexiyong** [comment]
梳理代码，看到if  pretrained这句，咋看感觉不对劲，还真有这bug啊

**Comment by bubbliiiing** [comment]
是的
